common/djangoapps/student/views.py

```diff
@@ -1464,13 +1473,38 @@ def create_account_with_params(request, params):
         extended_profile_fields=extended_profile_fields,
         enforce_username_neq_password=True,
         enforce_password_policy=enforce_password_policy,
-        tos_required=tos_required
+        tos_required=tos_required,
     )
 
     with transaction.commit_on_success():
-        ret = _do_create_account(form)
-
-    (user, profile, registration) = ret
+        # first, create the account
+        (user, profile, registration) = _do_create_account(form)
+
+        # next, link the account with social auth, if provided
+        if should_link_with_social_auth:
+            request.social_strategy = social_utils.load_strategy(backend=params['provider'], request=request)
+            social_access_token = params.get('access_token')
+            if not social_access_token:
+                raise ValidationError({
+                    'access_token': [
+                        _("An access_token is required when passing value ({}) for provider.").format(
+                            params['provider']
+                        )
+                    ]
+                })
+            request.session[pipeline.AUTH_ENTRY_KEY] = pipeline.AUTH_ENTRY_REGISTER_API
+            pipeline_user = None
+            error_message = ""
+            try:
+                pipeline_user = request.social_strategy.backend.do_auth(social_access_token, user=user)
+            except HTTPError:
+                error_message = _("The provided access_token is not valid.")
+            except AuthException as auth_exception:
```

Comment:
Do we have reason to believe that any of those can actually happen? AuthTokenRevoked definitely can't, because it's only ever raised by the `vk-openapi` backend. And do they all have messages that are appropriate for returning to the client?